src/main/kotlin/org/rust/cargo/commands/impl/CargoMetadata.kt

```diff
@@ -101,4 +107,61 @@ object CargoMetadata {
          */
         val dependencies: List<String>
     )
+
+    fun intoCargoProjectDescription(project: Project): CargoProjectDescription? {
+        val isRootPkg = { pkg: Package -> pkg.id == project.resolve.root }
+        val orderedPackages = project.packages.filter(isRootPkg) + project.packages.filter { !isRootPkg(it) }
```

Comment:
It was a awful idea not to remember the root package explicitly, and instead relay on the fact that the root package is first. Fixed.